packages/plugin-odoo/src/Odoo.ts

```diff
@@ -60,12 +61,9 @@ const imageClassesLabels = {
  * If more images are within the range, return undefined.
  */
 function getSingleImage(range: VRange): ImageNode | undefined {
-    const next = range.start.nextLeaf();
-    if (next instanceof ImageNode) {
-        const prev = range.end.previousLeaf();
-        if (prev === next) {
-            return next;
-        }
+    const selected = range.selectedNodes(node => node instanceof AtomicNode);
```

Comment:
I think this function was previously refactored by @Gorash for performance reasons. Further checking is needed to ensure your version is not reverting back to a less performant state.

Reply:
@Gorash @Gorash @Gorash

Reply:
no!
don't use selectedNodes in getSingleImage, it's called fto display button too.